modules/react/popup/stories/examples/CustomTarget.tsx

```diff
@@ -8,14 +8,18 @@ import {
   useInitialFocus,
   useReturnFocus,
 } from '@workday/canvas-kit-react/popup';
+import {useForkRef} from '@workday/canvas-kit-react/common'
 
 interface MyTargetProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   label: string;
 }
 
-const MyTarget = ({label, ...props}: MyTargetProps) => {
-  return <button {...props}>{label}</button>;
-};
+const MyTarget = React.forwardRef(({label, ...props}: MyTargetProps, ref) => {
+  const myRef = React.useRef(ref);
+	const elementRef = useForkRef(ref, myRef);
```

Comment:
Yessss! 👑

Reply:
I don't think we need to fork here. Using `React.forwardRef` will forward any ref assigned when our component is called. This means it might not be defined. If our component needs to use a `RefObject` (a `ref` with a `.current` value), _and_ we need to forward the ref, we must create our own and then fork it, passing the `elementRef` to the component in use.

Ref forking is also required anytime we need to merge 2 refs (a more accurate name might actually be `useMergeRef`, but the name was already solidified by material-ui). Popup hooks that deal with refs defined by the popup model also need to fork refs to make sure every ref instance points to the same value.

The following would be a reason to fork:

```tsx
const localRef = React.useRef<HTMLButtonElement>(null)
const elementRef = useForkRef(ref, localRef)

useMount(() => {
  // focus when mounted
  localRef.current.focus()
})

return <button ref={elementRef} />
```

This type of case is common and requires you to specify the element type `HTMLButtonElement`, even though it is already known. The following is exactly the same, but requires no Typescript overrides:

```tsx
const { localRef, elementRef } = useLocalRef(ref)
useMount(() => {
  // focus when mounted
  localRef.current.focus()
})

return <button ref={elementRef} />
```

My guess is this pattern was copied from the internal implementation of `Popup.Target`, which needs to merge the `model.state.targetRef` with a forwarded `ref`, which is not the case here.